requirements.txt

```diff
@@ -2,17 +2,10 @@ einops
 einops-exts
 transformers>=4.28.1
 torch==2.0.1
-torchvision
 pillow
-more-itertools
-datasets
-braceexpand
-webdataset
-wandb
-nltk
-scipy
-inflection
-sentencepiece==0.1.98
-pycocoevalcap
-pycocotools
-open_clip_torch>=2.16.0
+open_clip_torch>=2.16.0 
+
+# more-itertools  # not used anywhere
+# datasets  # doesn't appear to be used anywhere
+# scipy  # scipy does not appear to be used in this package
+# sentencepiece==0.1.98  # does not appear to be used
```

Comment:
I think sentencepiece is used under the hood for tokenizers. Can we add this back in?

Reply:
done

Reply:
Do we need to include scipy in the main dependencies? Also, do we need to depend on sentencepiece directly? Or can we do it indirectly through tokenizers. Ideally, if tokenizers is updated to remove sentencepiece as a dependency, we shouldn't continue including it in the distribution - right?

Reply:
Yep we can move scipy to eval requirements. Iirc sentencepiece was not installed by default in tokenizers but maybe this is no longer the case, I need to check again?

Reply:
> Yep we can move scipy to eval requirements. Iirc sentencepiece was not installed by default in tokenizers but maybe this is no longer the case, I need to check again?

It's here in `transformers` --
`"sentencepiece>=0.1.91,!=0.1.92",`
https://github.com/huggingface/transformers/blob/41c5f45bfe958be58bac6c891652f632ebac23e2/setup.py#L165